reference sheets (#29)
SciBorgs/SciGuides: reference-sheets/Sensors.md

They measure rotation. Many brushless motors have encoders integrated inside of them, while others can be found outside of the robot. Most measure what is effectively rotational displacement, as encoders read negative values from relative backwards movement.

All encoders have some sort of default unit. Various vendors will have different methods of changing the units returned; read their docs!

### [Relative Encoders](https://docs.wpilib.org/en/stable/docs/software/hardware-apis/sensors/sensor-overview-software.html#sensor-overview-software)

When powered on, its distance measurement will read 0. The zero-point will change on code boot, making its measurements "relative" to whenever it started.

### [Absolute Encoders](https://docs.wpilib.org/en/stable/docs/software/hardware-apis/sensors/encoders-software.html#encoders-software)

Has a set zero point. Will always know where it is, even between code deploys.

## [Gyroscope](https://docs.wpilib.org/en/stable/docs/software/hardware-apis/sensors/gyros-software.html)

It measures the rate of rotation of whatever plane it is on (and sometimes its relative axes). Usually found on the base of the drivetrain.

> ## [Beambreaks]()

Reviewer: empty link, honestly not sure what to link to here; maybe nothing at all

Detect if something has passed through them. Two key components: a part that shoots a ray of light, and a receiver. When the receiver no longer detects light, a signal is returned.

> For our program, beambreak sensors return true when unblocked, and false when blocked. **THIS IS SUBJECT TO CHANGE.**

Reviewer: ```suggestion
For our specific hardware, beambreak sensors return true when unblocked, and false when blocked.
```
